### PR TITLE
Re-enable sorbet

### PR DESCRIPTION
**Pull request by ivoanjo**
This PR re-enables the Sorbet type checker for the `lib/`, `spec/` and `benchmark/` folders.

It includes the following steps:
* Upgrading sorbet to the latest version
* Adding the sorbet `# typed: ...` comment to all files that were missing it
* Enabling sorbet typing for all files on which it currently passes
* Refresh the sorbet `rbi` files (which provide type definitions)

Apologies for the big diff; reviewing commit-by-commit should be doable 😅

### CODE REVIEW

File: lib/datadog/appsec.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# typed: false
```
Comment (lloeki) [inline lib/datadog/appsec.rb:1]: Should comment directives have an empty line separating them from these `require`?

File: lib/datadog/appsec/contrib/rails/ext.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# typed: true
```
Comment (lloeki) [inline lib/datadog/appsec/contrib/rails/ext.rb:1]: Should comment directives have an empty line separating them from the `module`/`class` items? Presumably having these non-separated makes the comment part of the module or class docstring.

**Review by lloeki — APPROVED** [review]
LGTM, left a couple of notes regarding comment directives, which may be better addressed in a separate PR to make it uniform across all files.

**Comment by ivoanjo** [comment]
Thanks!

I agree with you that not having the separation is really annoying. By default sorbet doesn't seem to have a way of doing it, so we'll need to do it manually, but that seems reasonable as we don't add dozens files to the project every day.

I'll merge this one and open a follow-up PR to add the newlines.

**Pull request merged**